plugins/cilium-cni/cmd/cmd.go

```diff
@@ -952,6 +953,92 @@ func (cmd *Cmd) Check(args *skel.CmdArgs) error {
 	return nil
 }
 
+// Status implements the cni STATUS verb.
+// Currently, it
+//   - checks cilium-cni connectivity with cilium-agent using healthz endpoint
+//   - If cilium-cni depends on delegated plugins (host-local, azure etc) for IPAM,
+//     cilium-cni invokes the STATUS API of the delegated plugins and passes the result back.
+func (cmd *Cmd) Status(args *skel.CmdArgs) error {
+	n, err := types.LoadNetConf(args.StdinData)
+	if err != nil {
+		return cniTypes.NewError(cniTypes.ErrInvalidNetworkConfig, "InvalidNetworkConfig",
+			fmt.Sprintf("unable to parse CNI configuration \"%s\": %v", string(args.StdinData), err))
+	}
+
+	if err := setupLogging(n); err != nil {
+		return cniTypes.NewError(cniTypes.ErrInvalidNetworkConfig, "InvalidLoggingConfig",
+			fmt.Sprintf("unable to setup logging: %s", err))
+	}
+
+	logger := loggerWithArguments(log.WithField(logfields.EventUUID, uuid.New()), args)
+
+	if n.EnableDebug {
+		if err := gops.Listen(gops.Options{}); err != nil {
+			log.WithError(err).Warn("Unable to start gops")
+		} else {
+			defer gops.Close()
+		}
+	}
+	logger.WithField("netconf", logfields.Repr(n)).Debugf("Processing CNI STATUS request")
+
+	if n.PrevResult != nil {
+		logger.WithField("previousResult", logfields.Repr(n.PrevResult)).Debugf("CNI Previous result")
+	}
+
+	cniArgs := &types.ArgsSpec{}
+	if err = cniTypes.LoadArgs(args.Args, cniArgs); err != nil {
+		return cniTypes.NewError(cniTypes.ErrInvalidNetworkConfig, "InvalidArgs",
+			fmt.Sprintf("unable to extract CNI arguments: %s", err))
+	}
+	logger = loggerWithCNIArgs(logger, cniArgs)
+
+	c, err := client.NewDefaultClientWithTimeout(defaults.ClientConnectTimeout)
+	if err != nil {
+		// use ErrTryAgainLater to tell the runtime that this is not a check failure
+		return cniTypes.NewError(cniTypes.ErrTryAgainLater, "DaemonDown",
+			fmt.Sprintf("unable to connect to Cilium agent: %s", client.Hint(err)))
+	}
+
+	if _, err := c.Daemon.GetHealthz(nil); err != nil {
```

Comment:
Move this to the else part of the chainAction if condition below ?. Currently, In chained mode, every call to Status() triggers two identical API calls to the agent because the same call is already being made within GenericVethChainer.